app/src/main/java/app/nexd/android/ui/seeker/create/SeekerCreateRequestViewModel.kt

```diff
@@ -1,47 +1,78 @@
 
 class SeekerCreateRequestViewModel(private val api: Api) : ViewModel() {
 
-    enum class State {
-        LOADING,
-        IDLE,
-        FINISHED
+    sealed class Progress {
+        object Idle : Progress()
+        class Error(@StringRes val message: Int) : Progress()
+        object Finished : Progress()
     }
 
-    private val state = BehaviorProcessor.createDefault(State.LOADING)
+    val progress = MutableLiveData<Progress>(Progress.Idle)
 
-    fun getCurrentUser(): LiveData<User> {
-        return LiveDataReactiveStreams.fromPublisher(
-            api.userControllerFindMe().toFlowable(BackpressureStrategy.LATEST)
-        )
-    }
+    val articles = LiveDataReactiveStreams.fromPublisher(
+        api.articlesControllerFindAll()
+            .map { articles ->
+                articles.map { HelpRequestCreateArticleBinder.ArticleInput(it) }
+            }
+            .toFlowable(BackpressureStrategy.BUFFER)
+    )
 
-    fun getArticles() : LiveData<List<Article>> {
-        return LiveDataReactiveStreams.fromPublisher(
-            api.articlesControllerFindAll().toFlowable(BackpressureStrategy.BUFFER)
-        )
-    }
+    private val compositeDisposable = CompositeDisposable()
 
-    fun sendRequest(request: HelpRequestCreateDto) {
-        if (request.articles.isNullOrEmpty()) {
-            return // request without articles shouldn't be accepted
-        }
-        with(api) {
-            helpRequestsControllerInsertRequestWithArticles(request)
-                .subscribe { state.onNext(State.FINISHED) }
-        }
-    }
+    fun sendRequest() {
+        val articles = this.articles.value
+            ?.filter { it.amount > 0 }
+            ?.map {
+                CreateHelpRequestArticleDto()
+                    .articleCount(it.amount)
+                    .articleId(it.article.id)
+            } ?: kotlin.run {
+                return
+            }
 
-    fun state() = LiveDataReactiveStreams.fromPublisher(state)
+        // TODO add additional information when in design
+        val disposable = api.userControllerFindMe()
+            .flatMap { currentUser ->
+                api.helpRequestsControllerInsertRequestWithArticles(
+                    HelpRequestCreateDto()
+                        .articles(articles)
+                        .street(currentUser.street)
+                        .number(currentUser.number)
+                        .zipCode(currentUser.zipCode)
+                        .city(currentUser.city)
+                        .phoneNumber(currentUser.phoneNumber)
+                )
+            }
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe({
+                progress.value = Progress.Finished
+            }, {
+                progress.value =
+                    Progress.Error(R.string.error_message_unknown) // TODO add proper error handling
+            })
 
+        compositeDisposable.add(disposable)
```

Comment:
we also have kotlinRx if you prefer to chain addToComposable() at the end, without having to store it in a local variable.

Reply:
Can you give me an example, how this would work?

Reply:
https://github.com/NexdApp/nexd-android/pull/104/files#diff-6902ed5c816a4866c9403a48e02f8ee8R67